### PR TITLE
Feature/wod 12 sell payment unit tests

### PR DESCRIPTION
**Pull request by LokmaneKrizou**

### CODE REVIEW

**Review by Fabio-Insolia-LiveStyled — APPROVED** [review]

File: sdk/src/test/java/com/realifetech/sdk/sell/payment/PaymentFeatureTest.kt [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+        paymentFeature.deletePaymentSource(PAYMENT_INTENT_ID) { error, response ->
+            assert(error is Exception)
+            assertEquals(ERROR_MESSAGE, error?.message)
+            assertEquals(null, response)
+        }
+    }
+
+
+    companion object {
+        private const val ERROR_MESSAGE = "Time to Crash baby!"
```
Comment (sjulia7) [inline sdk/src/test/java/com/realifetech/sdk/sell/payment/PaymentFeatureTest.kt:254]: :DD

**Review by sjulia7 — APPROVED** [review]

File: core-sdk/src/test/java/com/realifetech/core_sdk/feature/payment/repository/PaymentRepositoryTest.kt [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+        MockKAnnotations.init(this)
+        paymentRepository = PaymentRepository(paymentDataSource)
+        paymentIntentSlot = CapturingSlot()
+        paymentSourceSlot = CapturingSlot()
+        paginatedObjectSlot = CapturingSlot()
+    }
+
+    @Test
+    fun `add Payment Source returns Data`() {
+        //Given
```
Comment (mickeylee25) [inline core-sdk/src/test/java/com/realifetech/core_sdk/feature/payment/repository/PaymentRepositoryTest.kt:45]: Is it a standard rule to have these kinds of comments on Android?

File: core-sdk/src/test/java/com/realifetech/core_sdk/feature/payment/repository/PaymentRepositoryTest.kt [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+            paymentDataSource.addPaymentSource(paymentSourceInput, capture(paymentSourceSlot))
+        } answers {
+            paymentSourceSlot.captured.invoke(null, paymentSource.asModel)
+        }
+        //When
+        paymentRepository.addPaymentSource(paymentSourceWrapper) { error, response ->
+            //Then
+            Assert.assertEquals(null, error)
+            Assert.assertEquals(paymentSource.asModel, response)
+            print(response)
```
Comment (mickeylee25) [inline core-sdk/src/test/java/com/realifetech/core_sdk/feature/payment/repository/PaymentRepositoryTest.kt:56]: Is this for debugging or necessary for unit testing?

**Review by mickeylee25 — APPROVED** [review]

File: core-sdk/src/test/java/com/realifetech/core_sdk/feature/payment/repository/PaymentRepositoryTest.kt (hunk 2 above)
Comment (LokmaneKrizou) [reply core-sdk/src/test/java/com/realifetech/core_sdk/feature/payment/repository/PaymentRepositoryTest.kt:45]: no but i add them to make it easier to read.

**Pull request merged**